### PR TITLE
Simplify publishing releases to npm

### PR DESCRIPTION
**Pull request by cressie176**
I've added a GitHub action for automatically publishing releases to npm. The envisioned workflow is as follows...

1. Work normally, merging PRs, until ready to release etc
2. Check that the main branch build is green
3. Check that your local environment is clean and on main and up to date
4. Use `npm version major|minor|patch` to update package.json and create a tag, e.g. v0.8.1
5. Use `git push && git push --tags` to push both the package.json change and the new tag
6. Create a new release from the release in the GitHub UI
7. Wait for the new Publish action to publish the release to npm

For this to work we will need a repository secret configured with an NPM access token. It will be better to create this from a new npm robot account, as there's no way to restrict the access token to just the amqplib module.

One concern I have is that a GitHub action could be modified to expose the secret, which could then be used to publish a vulnerable version of amqplib. Since secrets are not exposed to forks, a maintainer would have to merge the malicious action first though. We should only accept this PR if we are happy with this risk.

### CODE REVIEW

File: .github/workflows/node-ci.yml [hunk 1]
```diff
@@ -14,7 +14,7 @@ jobs:
     runs-on: ubuntu-latest
     services:
       rabbitmq:
-        image: rabbitmq
+        image: rabbitmq:3.9-alpine
```
Comment (cressie176) [inline .github/workflows/node-ci.yml:17]: alpine should be faster to download. I also noticed some odd problems with the ubuntu version where it looked like ubuntu wasn't able to install rabbitmq.

File: .github/workflows/node-publish.yml [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    steps:
+      - uses: actions/checkout@v3
+        uses: actions/setup-node@v3
+        with:
+          node-version: '16.x'
+          registry-url: https://registry.npmjs.org/
+
+      - run: make lib/defs.js
+
+      - run: npm publish --dry-run
```
Comment (cressie176) [inline .github/workflows/node-publish.yml:22]: Will remove --dry-run after testing

File: .github/workflows/node-ci.yml (hunk 1 above)
Comment (kibertoad) [reply .github/workflows/node-ci.yml:17]: Any reason why you are not using the latest version?

Comment (cressie176) [reply .github/workflows/node-ci.yml:17]: Good spot, I'll change to 3-alpine.

Comment (kibertoad) [inline .github/workflows/node-ci.yml:17]:
```suggestion
        image: rabbitmq:3-alpine
```

File: .github/workflows/node-publish.yml [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+jobs:
+  publish:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v3
+        uses: actions/setup-node@v3
+        with:
+          node-version: '16.x'
```
Comment (kibertoad) [inline .github/workflows/node-publish.yml:17]:
```suggestion
          node-version: '18.x'
```

**Comment by kibertoad** [comment]
@cressie176 Do you need any help with this?

**Comment by cressie176** [comment]
Hi @kibertoad,

Unfortunately I don't have sufficient access to the GitHub repo to add a secret.

**Pull request closed without merging**